### PR TITLE
Add compatibility with SonataAdmin master branch

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

File: config/projects.yml [hunk 1]
```diff
@@ -214,11 +214,10 @@ doctrine-mongodb-admin-bundle:
 doctrine-orm-admin-bundle:
   branches:
     master:
-      php: ['7.2', '7.3', '7.4']
+      php: ['7.3', '7.4']
       target_php: '7.3'
       versions:
         symfony: ['4.4']
-        sonata_admin: ['3']
```
Comment (jordisala1991) [inline config/projects.yml:221]: maybe it is better to use 4 here

Comment (VincentLanglet) [reply config/projects.yml:221]: I don't know how this config works, but the version 4 doesn't exist.

Comment (VincentLanglet) [reply config/projects.yml:221]:
```
- php: '7.3'
      env: SONATA_ADMIN=4.*
```
will fail

Comment (jordisala1991) [reply config/projects.yml:221]: This works by adding an extra build for 4.x-dev of sonata admin. I guess now it is not required , but when you release 4, it is better to also test with unreleased 4.x (to ensure the latest Pr does not break compatibility)

Comment (VincentLanglet) [reply config/projects.yml:221]: Yes, but it does 2 things:
```
{% for package_name,package_versions in versions %}
{% for version in package_versions %}
    - php: '{{ target_php|default(php|last) }}'
      env: {{ package_name|upper }}={{ version }}.*
{% endfor %}
{% if package_versions|length > 0 %}
    - php: '{{ target_php|default(php|last) }}'
      env: {{ package_name|upper }}='dev-master as {{ package_versions|last }}.x-dev'
{% endif %}
{% endfor %}
```
Will add 
```
- php: '7.3'
   env: SONATA_ADMIN=4.*
```

And 
```
{% for package_name,package_versions in versions %}
{% if package_versions|length > 0 %}
    - env: {{ package_name|upper }}='dev-master as {{ package_versions|last }}.x-dev'
{% endif %}
{% endfor %}
```
Will add an extra build for 4.x-dev.

If we want the second, the first will be added and will obviously fail.
So we need to wait for the 4.x release in order to re-add this.

**Comment by VincentLanglet** [comment]
@OskarStark @greg0ire Can you merge this ? :)

File: config/projects.yml (hunk 1 above)
Comment (core23) [inline config/projects.yml:221]: Should we add `dev-master` to test against the master branch?

Comment (VincentLanglet) [reply config/projects.yml:221]: The composer.json is already using dev-master, this would be a duplicate build then.

**Comment by greg0ire** [comment]
Can you please improve your commit message to explain why and how this makes the orm bundle job suite compatible with the sonata admin master branch? Are both changes strictly related to that? It's not super obvious to me…

**Comment by VincentLanglet** [comment]
@greg0ire Is it better ?

**Comment by greg0ire** [comment]
It is, but why did you make 2 commits? One does not make sense without the other, right? Commit messages can have a body in case you don't know.

**Comment by VincentLanglet** [comment]
> It is, but why did you make 2 commits? One does not make sense without the other, right? Commit messages can have a body in case you don't know.

I thought you wanted two commit. Fixed.

**Comment by greg0ire** [comment]
Ok but now the commit messages is unhelpful… I think I understood what this is about, and changed it. Please tell me if that is indeed what you mean to do, and if yes, why we stop running tests with v3.

**Comment by VincentLanglet** [comment]
> Ok but now the commit messages is unhelpful… I think I understood what this is about, and changed it. Please tell me if that is indeed what you mean to do,

It is.

> and if yes, why we stop running tests with v3.

I just to on master branch.
SonataDoctrineORMAdmin v3 will still run test with SonataAdmin v3

But I'm making a PR to use SonataDoctrineORMAdmin dev-master and SonataAdmin dev-master https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1073

If I keep running test with SonataAdmin v3 (on master branch), it will try to use SonataDoctrineORMAdmin dev-master and SonataAdmin v3. This won't work.

**Comment by greg0ire** [comment]
Thanks @VincentLanglet !

**Pull request merged**